### PR TITLE
Added Search Component instead of having it in the NavBar component

### PR DESCRIPTION
**Pull request by khaledsaadeh**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,31 +4,24 @@ import MovieList from "./components/MovieList";
 import axios from "axios";
 import { data } from "autoprefixer";
 import NavBar from "./components/NavBar";
+import Search from "./components/Search";
 
-const searchAPI =
-  "https://api.themoviedb.org/3/search/movie?api_key=4f298a53e552283bee957836a529baec&language=en-US&page=1&include_adult=false&query=";
 const nowPlayingAPI =
```
Comment (MuhammadObeidat) [inline src/App.js:9]: This should be uppercase

File: src/components/Search.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+
+const Search = (props) => {
+  const searchAPI =
+    "https://api.themoviedb.org/3/search/movie?api_key=4f298a53e552283bee957836a529baec&language=en-US&page=1&include_adult=false&query=";
```
Comment (MuhammadObeidat) [inline src/components/Search.js:6]: Three notes here:
- if the constant you define is changing dynamically for a certain condition based on the props, then you should wrap it up with useMemo and inject this props to dependency array
-  if it's only a constant, just move it outside the component function declaration 
-  constant should be uppercase in most cases

File: src/components/Search.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  const [search, setSearch] = useState("");
+
+  const fetchSearch = () => {
+    axios.get(searchAPI, { params: { query: search } }).then((res) => {
+      props.setMovies(res.data.results);
+    });
+  };
```
Comment (MuhammadObeidat) [inline src/components/Search.js:14]: - This should be moved to the parent and should receive a parameter of type string containing the search text query the user is looking for 
- For better performance we use useCallback as memoization // please ignore this or do it, it's up to u

File: src/components/Search.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  const searchHandler = (event) => {
+    event.preventDefault();
+    if (search.trim() !== "") {
+      fetchSearch();
+      renderSearchResults(props.movies);
```
Comment (MuhammadObeidat) [inline src/components/Search.js:20]: No need for this function, please remove it  and move the fetchSearch as callback to parent then pass the local query state as parameter

File: src/components/Search.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    setSearch(event.target.value);
+  };
+
+  const renderSearchResults = (movies) => {
+    props.setMovies(movies);
+  };
+  return (
+    <form onSubmit={searchHandler}>
+      <input
+        className="border-2 border-gray-300 h-10 px-5 pr-16 rounded-lg text-sm focus:outline-2 m-2"
```
Comment (MuhammadObeidat) [inline src/components/Search.js:34]: It's preferred to move this set of classes into scss file to keep html more clean, so for instance you can give this input a class name then copy all of these classes and apply it inside scss file using @apply

File: src/App.js [hunk 6]
```diff
@@ -4,31 +4,39 @@ import MovieList from "./components/MovieList";
...
 
   return (
     <div className="App">
-      <NavBar searchAPI = {searchAPI} renderSearchResults={renderSearchResults} />
+      <NavBar>
+        <Search
+          setSearch={setSearch}
+          search={search}
+          fetchSearch={fetchSearch}
+        />
```
Comment (hazemturki) [inline src/App.js:38]:
```suggestion
        <Search
          onChangeQuery={setSearch}
          query={search}
          onSearch={fetchSearch}
        />
```

**Review by hazemturki — APPROVED** [review]

**Pull request merged**